src/components/Carousel.js

```diff
@@ -107,36 +107,30 @@ class Carousel extends Component {
         this.setupCarousel();
     }
 
-    componentWillReceiveProps(nextProps) {
-        if (nextProps.selectedItem !== this.state.selectedItem) {
-            this.updateSizes();
-            this.moveTo(nextProps.selectedItem);
-        }
-
```

Comment:
if `selectedItem` prop _is going to change_, and it doesn't match `this.state.selectedItem`… do thing